container/test-impl-base/src/main/java/org/jboss/arquillian/container/test/impl/RunModeUtils.java

```diff
@@ -47,31 +49,64 @@ private RunModeUtils() { }
     * @param testMethod
     * @return
     */
-   public static boolean isRunAsClient(Deployment deployment, Class<?> testClass, Method testMethod)
+   public static boolean isRunAsClient(Deployment deployment, TestClass testClass, Method testMethod)
    {
-      boolean runMethodAsClient = testMethod.isAnnotationPresent(RunAsClient.class);
-      boolean runClassAsClient = testClass.isAnnotationPresent(RunAsClient.class);
-
       boolean runAsClient = true;
-      if(deployment != null)
+      if (deployment != null)
       {
-         runAsClient =  deployment.getDescription().testable() ? false:true;
-         runAsClient =  deployment.isDeployed() ? runAsClient:true;
+         runAsClient = deployment.getDescription().testable() ? false : true;
+         runAsClient = deployment.isDeployed() ? runAsClient : true;
```

Comment:
But shouldn't it be an error if deployable artifact is after all not deployed? Or am I missing sth?